pages/api/paybutton/download/transactions/[paybuttonId].ts

```diff
@@ -0,0 +1,142 @@
+import moment from 'moment'
+import {
+  PRICE_API_DATE_FORMAT,
+  RESPONSE_MESSAGES,
+  DEFAULT_PAYBUTTON_CSV_FILE_DELIMITER,
+  PAYBUTTON_TRANSACTIONS_FILE_HEADERS,
+  DECIMALS,
+  SUPPORTED_QUOTES,
+  DEFAULT_QUOTE_SLUG,
+  SupportedQuotesType
+} from 'constants/index'
+import { TransactionWithAddressAndPrices, fetchTransactionsByPaybuttonId, getTransactionValueInCurrency } from 'services/transactionService'
+import { PaybuttonWithAddresses, fetchPaybuttonById } from 'services/paybuttonService'
+import { streamToCSV } from 'utils/files'
+import { setSession } from 'utils/setSession'
+import { NextApiResponse } from 'next'
+import { Decimal } from '@prisma/client/runtime'
+
+export interface TransactionFileData {
+  amount: Decimal
+  date: moment.Moment
+  value: number
+  rate: number
+  transactionId: string
+  currency: string
+  address: string
+}
+
+export interface FormattedTransactionFileData {
+  amount: string
+  date: string
+  value: string
+  rate: string
+  transactionId: string
+  address: string
+}
+
+function isCurrencyValid (currency: SupportedQuotesType): boolean {
+  return SUPPORTED_QUOTES.includes(currency)
+}
+
+const getPaybuttonTransactionsFileData = (transaction: TransactionWithAddressAndPrices, currency: SupportedQuotesType): TransactionFileData => {
+  const { amount, createdAt, hash, address } = transaction
```

Comment:
`createdAt` won't necessarily reflect the true date the tx was sent/created.

When testing this PR, every tx in the CSV got today's date as their date since the txs have just been inserted to the DB when I reset the containers.

So to fix this, use the `timestamp` field, convert from unix timestamp to moment with `moment(timestamp * 1000)` and it should be good. 

For that, we have the field `timestamp` (which is a number for the

Reply:
Nice catch; missed this one - would have figured this out pretty quick after using it.

Reply:
> `createdAt` won't necessarily reflect the true date the tx was sent/created.
> 
> When testing this PR, every tx in the CSV got today's date as their date since the txs have just been inserted to the DB when I reset the containers.
> 
> So to fix this, use the `timestamp` field, convert from unix timestamp to moment with `moment(timestamp * 1000)` and it should be good.

thanks, addressed